### PR TITLE
[Improve][flink][Connector-V1] Improve druid and doris connector

### PR DESCRIPTION
**Pull request by enterwhat**
https://github.com/apache/incubator-seatunnel/issues/2893

feat #2893 :
(1), solve the problem that the -i parameter cannot be passed in the placeholder of the source part of the configuration file that replaces the conf, and supports placeholder replacement like the sql part.

(2), solve druid's start_date, end_date through the command line -i to pass the parameter space, recognized as the parameter separator of the shell command line, so that the date format parameter of yyyy-MM-dd HH:mm:ss cannot be recognized as a whole, resulting in running Errors, use specific placeholders by default, can be customized

(3), customize the output separator of druid to support importing into doris

(4), replace the default delimiter of doris to solve the problem that " "(\t) in the column value affects column recognition.

(5), druid adds user, password parameter authentication support

(6), provide provided annotation for the package that the druid connector can rely on seatunnel

### CODE REVIEW

**Comment by EricJoy2048** [comment]
Hi, @enterwhat . Thank's for your contribution. Please change the pr comment to english.

**Comment by enterwhat** [comment]
> 

done

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java [hunk 1]
```diff
@@ -169,6 +179,8 @@ private RowTypeInfo getRowTypeInfo(String jdbcURL, String datasource, Collection
             names[i] = field;
             i++;
         }
+        long count = Arrays.stream(typeInformation).filter(t -> null == t).count();
+        System.out.println("typeInformation null:::" + count);
```
Comment (hailin0) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java]: remove?

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/command/FlinkApiTaskExecuteCommand.java [hunk 2]
```diff
@@ -63,7 +61,10 @@ public FlinkApiTaskExecuteCommand(FlinkCommandArgs flinkCommandArgs) {
     public void execute() throws CommandExecuteException {
         EngineType engine = flinkCommandArgs.getEngineType();
         Path configFile = FileUtils.getConfigPath(flinkCommandArgs);
-
+        this.flinkCommandArgs.getVariables().stream()
+                .map(variable -> variable.split("=", 2))
+                .filter(pair -> pair.length == 2)
+                .forEach(pair -> System.setProperty(pair[0], pair[1]));
```
Comment (hailin0) [inline seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/command/FlinkApiTaskExecuteCommand.java]: @EricJoy2048 

please confirm

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java [hunk 3]
```diff
@@ -117,17 +117,27 @@ public CheckResult checkConfig() {
...
+        if (StringUtils.isBlank(escape_delimiter)){
+            escape_delimiter = ESCAPE_DELIMITER_DEFAULT;
+        }
+        String startTimestamp = config.hasPath(START_TIMESTAMP) ? config.getString(START_TIMESTAMP).replaceAll(escape_delimiter, " ") : null;
+        String endTimestamp = config.hasPath(END_TIMESTAMP) ? config.getString(END_TIMESTAMP).replaceAll(escape_delimiter, " ") : null;
         List<String> columns = config.hasPath(COLUMNS) ? config.getStringList(COLUMNS) : null;
-
+        System.out.println(jdbcURL);
+        System.out.println(user);
+        System.out.println(password);
```
Comment (hailin0) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java]: remove?

**Comment by hailin0** [comment]
image
https://github.com/apache/incubator-seatunnel/actions/runs/3132748547/jobs/5086755036

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/command/FlinkApiTaskExecuteCommand.java (hunk 2 above)
Comment (enterwhat) [reply seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/command/FlinkApiTaskExecuteCommand.java]: done

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java (hunk 1 above)
Comment (enterwhat) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java]: done

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java (hunk 3 above)
Comment (enterwhat) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java]: done

**Comment by enterwhat** [comment]
> Hi, @enterwhat . Thank's for your contribution. Please change the pr comment to english.

I have commit new commits, should I push a new pr?

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java (hunk 1 above)
Comment (enterwhat) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java]: I

> 

I have commit new commits, should I push a new pr?

Comment (EricJoy2048) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java]: > 

I suggest you don't push a new pr if the new pr and this pr have many code dependency. Of course, if there is no code dependency between two prs, you can submit.

Comment (enterwhat) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java]: > 

any problem with this pr? how long will this be reviewed again?

**Comment by enterwhat** [comment]
> image
> 
> https://github.com/apache/incubator-seatunnel/actions/runs/3132748547/jobs/5086755036

I have fixed this

**Comment by enterwhat** [comment]
> Hi, @enterwhat . Thank's for your contribution. Please change the pr comment to english.

any problem with this pr? how long will this be reviewed again?

**Comment by CalvinKirs** [comment]
hi, thanks for your contribution, dev branch will no longer maintain the V1 version of the connector, so I will close this PR,
If you are interested in this, you can contribute the corresponding V2Connector
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-connectors-v2

**Pull request closed without merging**